crates/client/db/src/l1_db.rs

```diff
@@ -128,4 +129,32 @@ impl MadaraBackend {
         self.db.put_cf_opt(&nonce_column, bincode::serialize(&nonce)?, /* empty value */ [], &writeopts)?;
         Ok(())
     }
+
+    pub fn get_l1_handler_tx_hashes(&self, l1_tx_hash: TxHash) -> Result<Vec<Felt>, DbError> {
+        let l1_l2_mappings_column = self.db.get_column(Column::L1MessagingHandlerTxHashes);
+        let mut l1_handler_tx_hashes = vec![];
+        for kv_bytes in self.db.prefix_iterator_cf(&l1_l2_mappings_column, l1_tx_hash.as_slice()) {
+            let l1_handler_tx_hash = Felt::from_bytes_be_slice(&kv_bytes?.1);
+            l1_handler_tx_hashes.push(l1_handler_tx_hash);
+        }
```

Comment:
Consider replacing this with a `collect` on `self.db.prefix_iterator_cf` to mitigate resizing incurred by `Vec::push`.